guides/introduction.md

```diff
@@ -0,0 +1,74 @@
+# Joken Overview
+
+[![Documentation](https://img.shields.io/badge/docs-hexpm-blue.svg)](http://hexdocs.pm/joken/) [![Downloads](https://img.shields.io/hexpm/dt/joken.svg)](https://hex.pm/packages/joken) [![Build](https://travis-ci.org/bryanjos/joken.svg?branch=master)](https://travis-ci.org/bryanjos/joken)
+
+[Documentation](http://hexdocs.pm/joken/)
+
+Joken is a JWT (Json Web Token) library based upon the awesome [`erlang-jose`](https://github.com/potatosalad/erlang-jose/). Features:
+
+  - Ease of key configuration. We provide built-in support with Elixir's `Mix.Config` system. See our configuration guide for more details.
+  - Portable configuration by leveraging `Joken.Config`. 
+  - Built-in claim generation and validation. `erlang-jose` is responsible only for signing and verifying token signatures. Here we provide extra tools for claim validation and generation.
+  - Better error handling. We provide `ExUnit` like error messages for claim validation. 
+  - A good perfomance analysis for ensuring this hot-path in APIs won't be your bottleneck. Please see our perfomance documentation to check what we are talking about.
+  - Good defaults. Joken comes with chosen good defaults for parsing Json and generating claims.
+  - Hooks for extending Joken functionality. All core actions in Joken have a corresponding hook for extending its functionality. See our hooks guide.
+  
+## JWT algorithms
+
+Joken supports all algorithms that JOSE supports. That contains:
+
+  - All HS, RS, ES, PS signing algorithms.
+  - All Edwards algorithms (Ed25519, Ed25519ph, Ed448, Ed448ph)
+  
+See [jose JWS algorithm support](https://github.com/potatosalad/erlang-jose#json-web-signature-jws-rfc-7515) for more information.
+
+## Usage
+
+As easy as:
+
+1. A key configuration:
+
+``` elixir
+# config/dev.exs 
+config :joken, default_signer: "secret"
```

Comment:
Given the recent talk about configuration, do we need to take anything in consideration when defining configs here? https://hexdocs.pm/elixir/master/library-guidelines.html#avoid-application-configuration

Reply:
My opinion is that with the latest changes it is already following the guide. I mean: we accept a signer as a parameter already in every possible function call we'd need a signer. So, currently, it is there more as convenience than anything.

We can maybe think about some initialization setup like phoenix and ecto do since we already have an application implementation. I'd add it to 2.1 though for not blocking this further.

I'll work on hook options then! :)